### ISSUE TITLE
[Feature Request] Session progress/goal window

### ISSUE THREAD

**Issue by nephlm** [olivierkes/manuskript#459]
Have tool option that opens a small non-modal which shows the wordcount for the current session and lets the user set a session goal, as well as display the current draft word count and it's progress toward the goal.  

This was called "Project Targets" in  scrivener.  I currently have a script that reads the disk every ten seconds or so to do this, but it would be nice to get something integrated.  

Something like this is what I'm talking about.

![screenshot at 2019-01-21 23 09 45](https://user-images.githubusercontent.com/8230238/51512058-f65a6380-1dd1-11e9-949f-ffb654ed4d8a.png)

The `Reset` button would reset the session word count to 0.

I have this mostly working with just a few nits to iron out, but this is my first Qt programming so there's that.

**Comment by gedakc** [comment]
Were you aware of the word target at the bottom of the **Editor** window?

![manuskript-editor-word-count](https://user-images.githubusercontent.com/10405019/51556365-399aed80-1e37-11e9-9d0e-262c0917e965.gif)

**Comment by nephlm** [comment]
Yes, but my understanding is it is connected to what is being shown in the main pane/edited.  

I was looking for session targets which may span many scenes.  (e.g. I'm sitting down to work and I have a goal of knocking out 1000 words).  That word count may be broken up across multiple scenes and chapters.   A scene goal represents how long the scene should be in the manuscript, the session target is how much work I want to get done (today, before lunch, whatever) without regard to where those words end up in the finished manuscript. 

I had written a hacky script that did it by reading the save file on disk every ten seconds or so, but I thought it would nice to have integrated (and less hacky).  It was a feature of scrivener that I had open pretty much all the time.

**Comment by gedakc** [comment]
Your understanding is correct.  If one wanted to see the progress over a writing session one could navigate to the top of the project (to include all chapters and scenes) and note the word count.  Then at the end of the writing session one could again navigate to the top of the project and note the latest word count.  The difference between the two numbers would indicate the writing session progress.

With that said I can understand your desire for the tracking method you are proposing.  However I am not a fan of having another window open.  I will leave the decision on whether to include such a feature with the author of Manuskript @olivierkes.

**Comment by tildagail** [comment]
Attempt at some feedback on this idea, from a user:

From the perspective of a writer using Manuskript, I like the session wordcount. For events like NaNoWriMo with a daily target, the at-a-glance "I've written ... words today!" is a great idea.

But, for me the wordcount of the current draft is not the wordcount of the project as a whole - I usually have a couple of versions in separate folders, plus deleted scenes. And, I don't always set wordcount goals. Draft wordcount should be optional and maybe movable (i.e. can be set to a specific folder).

**Comment by nephlm** [comment]
I originally had the draft target settable like the session target is, but I took it out for a few reasons.  First it wasn't clear what should happen if the user changed one of the scene targets and the other is that there already seems to be some support for the idea of an arbitrary target not derived from the scene targets.  

Specifically in the `Outline` Enum is the `setGoal` value, which has the comment `The goal set by the user, if any. Can be different from goal which can be computed (sum of all sub-items' goals)`.  But it's really not clear to me that this can be set for the root item in the UI.  In any case there seemed to be a system in place and I didn't want to build another one. 

I originally intended to make Draft and/or Session numbers hidable, so I'd prefer that over the selecting a folder at least for now.  Unfortunately a foldable container doesn't seem to be built into Qt5 and when it came to building a custom widget... I figured it wasn't needed for minimal viable window and I'd see if there was interest.

**Comment by russell-phillips** [comment]
Just commenting to note that I'd like to see a session word count, but I think my preference would be for it to be displayed in the status bar. A menu item to open a window showing session word count would also work, but I'd rather not have a window open permanently.

**Comment by estherflails** [comment]
I really like the idea of this feature. :) I'd just like to add that it would also be nice to be able to look at previous sessions, to see your daily progress. Perhaps it could be added to the left side menu, something similar to Outline, with "Date, Session, Word Count, Goal, %" columns?

**Comment by TheJackiMonster** [comment]
I've merged #814 today. So the next release will include a session target dialog. It does not include a statistics page as @estherflails made a mockup for. But I'll keep this issue here open. So maybe someone will add such an overview later.
